sw/device/otp_img/meson.build

```diff
@@ -0,0 +1,25 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+foreach device_name, device_lib : sw_lib_arch_core_devices
```

Comment:
Are you planning to add simulation platform specific switches to your script? If not then I don't see a need to iterate here.

Reply:
Also, I wanted to understand the motivation to put this in meson. Its because ALL chip level tests (even non SW ones) will need this image to be preloaded - is that correct?

Reply:
So for the time being, these images remain identical. 
But depending on how the multiple-tops situation evolves, we may run into a case where certain configurations in the HW_CFG or SW_CFG partitions need to be different among the platforms.
Hence, I thought it's probably best if we keep this flexibility here. 

We're not using this right now, however, so I would also be fine with removing it (just note that any vmem file naming scheme change has to be propagated through the documentation).

Reply:
Actually I looked at subsequent commits - because of the way the DV flow is set, it expects the image name to be suffixed with the simulation platform type, so this is needed.